### PR TITLE
Integrate marlin and poly commit as crates of ginger-lib

### PR DESCRIPTION
**Pull request by nicholas-mainardi**
Included marlin and polycommit inside proof-systems, to avoid dependencies headaches.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -35,10 +35,12 @@ script:
 jobs:
   include:
     - name: bionic_rust-1.51.0_latest-CHECK
-      env: IMAGE_TAG=bionic_rust-1.51.0_latest TESTS='000,001,003,005,006,008,009,011,012,013,014,015,016,020'
+      env: IMAGE_TAG=bionic_rust-1.51.0_latest TESTS='000,001,003,005,006,008,009,011,012,014,015,016,017,018,019,020,021,026'
     - name: bionic_rust-1.51.0_latest-TEST
```
Comment (95DDB) [inline .travis.yml]: Let's rename this "TEST_BACKEND"

File: ci/tests/022-cargo-test.sh [hunk 2]
```diff
@@ -3,4 +3,4 @@
 
 set -xeo pipefail
 
-cargo $CARGOARGS test --workspace --all-features --exclude "r1cs-std"
+cargo $CARGOARGS test --workspace --all-features --exclude "r1cs-std" --exclude "proof-systems"
```
Comment (95DDB) [inline ci/tests/022-cargo-test.sh]: Can't you pass a list here ?

File: bench-utils/src/lib.rs [hunk 3]
```diff
@@ -88,7 +88,7 @@ pub mod inner {
                 end_info,
                 message,
                 final_time,
-                pad = 75 - indent_amount
+                pad = 75 - indent_amount.min(75)
```
Comment (95DDB) [inline bench-utils/src/lib.rs:91]: Would be best to fix all the timers to make sure we close them in error case. Let's open an issue maybe

**Review by 95DDB — CHANGES_REQUESTED** [review]

**Comment by 95DDB** [comment]
@cronicc if you have time to take a look for what concerns you, though not mandatory :)

File: .travis.yml (hunk 1 above)
Comment (nicholas-mainardi) [reply .travis.yml]: Ok

File: ci/tests/022-cargo-test.sh (hunk 2 above)
Comment (nicholas-mainardi) [reply ci/tests/022-cargo-test.sh]: It seems not to be possible, the docs just mention that [`--exclude` can be employed multiple times](https://doc.rust-lang.org/cargo/commands/cargo-test.html#package-selection); the only other way mentioned in the docs would be relying on glob patterns, but they do not support `|` regular expression operator

**Review by 95DDB — APPROVED** [review]

File: bench-utils/src/lib.rs (hunk 3 above)
Comment (nicholas-mainardi) [reply bench-utils/src/lib.rs:91]: I opened issue #188

**Review by cronicc — APPROVED** [review]
LGTM

**Pull request merged**